### PR TITLE
Cookie tracking banner, cookies page, and analytics opt-out logic

### PR DESCRIPTION
**Pull request by WolfyUK**
### Context
[AB#218416](https://dfe-ssp.visualstudio.com/a14e55df-4fbf-4a2f-a11d-22b187178343/_workitems/edit/218416) [AB#218415](https://dfe-ssp.visualstudio.com/a14e55df-4fbf-4a2f-a11d-22b187178343/_workitems/edit/218415) [AB#218460](https://dfe-ssp.visualstudio.com/a14e55df-4fbf-4a2f-a11d-22b187178343/_workitems/edit/218460) [AB#218500](https://dfe-ssp.visualstudio.com/a14e55df-4fbf-4a2f-a11d-22b187178343/_workitems/edit/218500)

### Change proposed in this pull request
- Cookie banner with support to opt-out of tracking cookies that is only displayed if JavaScript it enabled. 
- Cookie page updates, including form for changing options even if JavaScript is disabled, and move to its own Controller as no longer static. 
- E2E and Integration test updates.

### Guidance to review 
To test locally ensure that a `APPINSIGHTS_INSTRUMENTATIONKEY` value is present in environment variables.

### Checklist
- [x] Work items have been linked [(use AB#)](https://learn.microsoft.com/en-us/azure/devops/boards/github/link-to-from-github?view=azure-devops#use-ab-to-link-from-github-to-azure-boards-work-items)
- [x] Your code builds clean without any errors or warnings
- [x] You have run all unit/integration tests and they pass
- [x] Your branch has been rebased onto main
- [x] You have tested by running locally

### CODE REVIEW

File: web/src/Web.App/Controllers/StaticContentController.cs [hunk 1]
```diff
@@ -35,7 +35,28 @@ public class StaticContentController : Controller
...
+        var cookiePolicy = HttpContext.Request.Cookies[Constants.CookieSettingsName];
+        var vm = new StaticCookiesViewModel(Constants.CookieSettingsName, cookiePolicy != "disabled");
+        return View(vm);
+    }
+
+    [HttpPost]
+    [Route("cookies")]
+    public IActionResult SaveCookies([FromForm(Name = "cookies-analytics")] bool analytics)
+    {
+        HttpContext.Response.Cookies.Append(Constants.CookieSettingsName, analytics ? "enabled" : "disabled", new CookieOptions
```
Comment (jrabbott) [inline web/src/Web.App/Controllers/StaticContentController.cs:48]: The cookie will need to be set to Secure & HttpOnly

File: web/src/Web.App/Views/Shared/_Layout.cshtml [hunk 2]
```diff
@@ -142,7 +141,8 @@
                         class="govuk-footer__link"
                         href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
                         rel="license">
-                        Open Government Licence v3.0</a>, except where otherwise stated
+                        Open Government Licence v3.0
```
Comment (WolfyUK) [inline web/src/Web.App/Views/Shared/_Layout.cshtml:144]: Arg. Why does this line break always get added?!

File: web/src/Web.App/Views/Shared/Components/Cookies/Default.cshtml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        </div>
+        <div class="govuk-button-group">
+            <button type="button" class="govuk-button hide-cookies" data-module="govuk-button">
+                Hide cookie message
+            </button>
+        </div>
+    </div>
+</div>
+
+<script type="text/javascript" add-nonce="true">
```
Comment (jrabbott) [inline web/src/Web.App/Views/Shared/Components/Cookies/Default.cshtml:55]: Can this be added to the script section?

File: web/src/Web.App/Views/Shared/Components/Cookies/Default.cshtml
Comment (jrabbott) [inline web/src/Web.App/Views/Shared/Components/Cookies/Default.cshtml:1]: How does this behave when JS is disabled?

File: web/src/Web.App/Views/Shared/Components/Analytics/Default.cshtml [hunk 4]
```diff
@@ -33,6 +38,15 @@
             {
                 @:sdk.setAuthenticatedUserContext("@User.UserGuid()");
             }
+            _sdk = sdk;
         },
     });
+    if (window !== undefined) {
+        window.setAppInsightsCookieEnabled = function (enabled) {
+            if (!_sdk) return;
+            console.debug("isEnabled was", _sdk.getCookieMgr().isEnabled());
```
Comment (jrabbott) [inline web/src/Web.App/Views/Shared/Components/Analytics/Default.cshtml:47]: Console 👀

File: web/src/Web.App/Views/Shared/Components/Cookies/Default.cshtml (hunk 3 above)
Comment (WolfyUK) [reply web/src/Web.App/Views/Shared/Components/Cookies/Default.cshtml:55]: Wrapping in `@section scripts` doesn't seem to render anything at all from a component view

File: web/src/Web.App/Views/Shared/Components/Cookies/Default.cshtml
Comment (WolfyUK) [reply web/src/Web.App/Views/Shared/Components/Cookies/Default.cshtml:1]: Wrapped in the same `<form>` as the cookie page

Comment (jrabbott) [reply web/src/Web.App/Views/Shared/Components/Cookies/Default.cshtml:1]: Probably need to regression test pages that have form elements, to ensure neither impacts the other?

Comment (WolfyUK) [reply web/src/Web.App/Views/Shared/Components/Cookies/Default.cshtml:1]: Agreed; in progress.

Comment (WolfyUK) [reply web/src/Web.App/Views/Shared/Components/Cookies/Default.cshtml:1]: No longer relevant

**Review by jrabbott — APPROVED** [review]

**Pull request merged**